### PR TITLE
fixes student fields and prepares for sparse config props

### PR DESCRIPTION
**Pull request by esotrope**

### CODE REVIEW

File: admin-service/src/main/java/org/opentestsystem/rdw/admin/multitenant/configprops/DefaultConfigpropsService.java [hunk 1]
```diff
@@ -58,9 +58,9 @@ public DefaultConfigpropsService(final ConfigpropsConsolidationService configpro
         if (set.size() > 1) {
             logger.info("More than one ReportingSystemSettings resolved, please ensure all tenant aware default configurations are identical, count[{}] {}", set.size(), set);
             //TODO: workaround until fix for RP-548
-            if (reportingSystemSettings.containsKey("reporting-webapp") && reportingSystemSettings.containsKey("reporting-service")) {
-                ReportingSystemSettings webappSettings = reportingSystemSettings.get("reporting-webapp");
-                ReportingSystemSettings reportingServiceSettings = reportingSystemSettings.get("reporting-service");
+            if (reportingSystemSettings.containsKey("rdw-reporting-webapp") && reportingSystemSettings.containsKey("rdw-reporting-service")) {
```
Comment (esotrope) [inline admin-service/src/main/java/org/opentestsystem/rdw/admin/multitenant/configprops/DefaultConfigpropsService.java]: allows the student fields to get thru

File: webapp/src/main/webapp/src/app/admin/tenant/component/property-override-tree-table/property-override-tree-table.component.ts [hunk 2]
```diff
@@ -96,8 +102,7 @@ function rowTrackBy(index: number, { node }: any): string {
   selector: 'property-override-tree-table',
   templateUrl: './property-override-tree-table.component.html',
   styleUrls: ['./property-override-tree-table.component.less'],
-  // taking this off so markAsDirty() works
-  // changeDetection: ChangeDetectionStrategy.OnPush,
+  changeDetection: ChangeDetectionStrategy.OnPush,
```
Comment (esotrope) [inline webapp/src/main/webapp/src/app/admin/tenant/component/property-override-tree-table/property-override-tree-table.component.ts]: was able to get this slight performance boosting setting back

File: webapp/src/main/webapp/src/app/admin/tenant/component/tenant-form/tenant-form.component.html [hunk 3]
```diff
@@ -274,13 +274,13 @@ <h1 class="h2" heading>
...
-        [open]="configurationOpen"
-        (openChange)="configurationOpen = $event"
+        [open]="configurationsOpen$ | async"
+        (openChange)="configurationsOpen$.next($event)"
         openLabel="{{ 'tenant-form.field.configurations.label' | translate }}"
         closedLabel="{{ 'tenant-form.field.configurations.label' | translate }}"
       ></app-toggle-label>
 
-      <div class="form-group" [hidden]="!configurationOpen">
+      <div class="form-group" [hidden]="!(configurationsOpen$ | async)">
```
Comment (esotrope) [inline webapp/src/main/webapp/src/app/admin/tenant/component/tenant-form/tenant-form.component.html]: fixes bug with config section not opening when form is submitted with errors in the section

File: webapp/src/main/webapp/src/app/admin/tenant/component/tenant-form/tenant-form.component.ts [hunk 4]
```diff
@@ -154,25 +157,25 @@ function toPropertiesProvider<T = any>(
...
+                    value.toLowerCase().includes(caseInsensitiveSearch)) ||
+                  (Object(value) !== value &&
+                    String(value)
+                      .toLowerCase()
+                      .includes(caseInsensitiveSearch)))) &&
+              (!modified || value !== defaultValue) &&
+              (!required || fieldRequired(controlKey))
+            );
+          })
+          .sort(byKey)
```
Comment (esotrope) [inline webapp/src/main/webapp/src/app/admin/tenant/component/tenant-form/tenant-form.component.ts]: added sorting of the entries by key

File: webapp/src/main/webapp/src/app/admin/tenant/component/tenant-form/tenant-form.component.ts [hunk 5]
```diff
@@ -414,7 +420,9 @@ export class TenantFormComponent implements OnChanges, OnDestroy {
           );
           const flattened = keyBy(properties, ({ key }) => key);
           const unflattened = unflatten(flattened);
-          return toTreeNodes(unflattened, hasSearch || invalid);
+          // was based on having a search and if the form was submitted with errors but i think it is easiest if it
+          // is just always open when the config section is expanded.
+          return toTreeNodes(unflattened, true /*hasSearch || invalid*/);
```
Comment (esotrope) [inline webapp/src/main/webapp/src/app/admin/tenant/component/tenant-form/tenant-form.component.ts]: just made the table show all by default i found it a pain to drill down - i'll leave it up for a vote though - only downside i saw with this is it lags a little to filter when it starts with everything showing

File: webapp/src/main/webapp/src/app/admin/tenant/model/tenants.ts [hunk 6]
```diff
@@ -135,22 +136,23 @@ export function toTenant(
...
-          aggregate,
-          reporting
-        }
-      : {
-          archive,
-          datasources,
-          aggregate,
-          reporting
-        },
+  // adds in all the missing fields from the sparse set provided by the backend
```
Comment (esotrope) [inline webapp/src/main/webapp/src/app/admin/tenant/model/tenants.ts]: @ftdholbrook the prep for your change - tested this too

Comment (ftdholbrook) [reply webapp/src/main/webapp/src/app/admin/tenant/model/tenants.ts]: @esotrope if there are some fields inside those top level objects with null values would that cause those fields to disapear?

File: webapp/src/main/webapp/src/app/admin/tenant/model/tenants.spec.ts [hunk 7]
```diff
@@ -5,43 +5,43 @@ import { TenantConfiguration } from './tenant-configuration';
...
-        'TENANT'
-      )
-    ).toEqual({
-      archive: {},
-      aggregate: {},
-      datasources: {},
-      reporting: {}
-    });
-  });
+  // it('should prune based on tenant type SANDBOX', () => {
```
Comment (esotrope) [inline webapp/src/main/webapp/src/app/admin/tenant/model/tenants.spec.ts]: going to re-enable these once the expectations of the api behavior stabalize and i can copy some network json responses

File: webapp/src/main/webapp/src/app/admin/tenant/model/tenants.ts (hunk 6 above)
Comment (esotrope) [reply webapp/src/main/webapp/src/app/admin/tenant/model/tenants.ts]: yes it should.

Comment (esotrope) [reply webapp/src/main/webapp/src/app/admin/tenant/model/tenants.ts]: sorry misread. no it wont disappear

Comment (esotrope) [reply webapp/src/main/webapp/src/app/admin/tenant/model/tenants.ts]:
```
const a = { a: null, b: null };
const b = { a: 1 };
const c = { ...a, ...b } // { a: 1, b: null}
```

**Pull request merged**